### PR TITLE
feat(frontend): reusable validation schema

### PR DESCRIPTION
**Pull request by sebastien-comeau**
This commit introduces a robust string validation schema, `stringSchema`, along with new validation schemas for first name, last name, and email address. These schemas leverage Zod for type safety and offer customizable options including:

- **String Length Validation:** Enforce minimum and maximum string lengths.
- **Format Validation:** Predefined formats (`alpha-only`, `alphanumeric`, `digit-only`, `non-digit`) and custom regular expression support.
- **Trimming:** Automatic trimming of leading/trailing whitespace (configurable).
- **Customizable Error Messages:** Tailor error messages for each validation rule.
- **100% Test Coverage:** Comprehensive unit tests ensure the reliability and correctness of the validation logic, covering all features and edge cases.

The new schemas (`firstNameSchema`, `lastNameSchema`, and `emailAddressSchema`) build upon the flexible `stringSchema` to provide specialized validation for common use cases. This improves data integrity and provides a consistent user experience.

### CODE REVIEW

File: frontend/app/.server/validation/email-address-validation-schema.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    maxLength: 'Email address must be less than or equal to {{length}} characters.',
+    required: 'Email address is required.',
+  },
+  fr: {
+    format: "Le format de l'adresse courriel est invalide.",
+    invalidType: "L'adresse courriel doit être une chaîne de caractères.",
+    maxLength: "L'adresse courriel doit contenir au maximum {{length}} caractères.",
+    required: "L'adresse courriel est requis.",
+  },
+} as const satisfies Record<Language, Required<EmailAddressValidationSchemaErrorMessages>>;
```
Comment (gregory-j-baker) [inline frontend/app/.server/validation/email-address-validation-schema.ts:37]: The validators should return i18n keys which are resolved at render time by the i18n subsystem.

Comment (sebastien-comeau) [reply frontend/app/.server/validation/email-address-validation-schema.ts]: I removed the localization for this pull request for now.

File: frontend/app/.server/validation/email-address-schema.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  errorMessages?: EmailAddressSchemaErrorMessages;
+  maxLength?: number;
+}
+
+const DEFAULT_MESSAGES = {
+  format_error: 'Email address format is invalid.',
+  invalid_type_error: 'Email address must be a string.',
+  max_length_error: 'Email address must be less than or equal to {maximum} characters.',
+  required_error: 'Email address is required.',
+} as const satisfies Required<EmailAddressSchemaErrorMessages>;
```
Comment (gregory-j-baker) [inline frontend/app/.server/validation/email-address-schema.ts:48]: These should be i18n keys. Check out how Java does it: ><https://javaee.github.io/javaee-spec/javadocs/javax/validation/constraints/Email.html#message-->

Comment (sebastien-comeau) [reply frontend/app/.server/validation/email-address-schema.ts:48]: I disagree. Zod will always return issues with the path and code and default formatted message. It let  you override default error message like I did.

Comment (sebastien-comeau) [reply frontend/app/.server/validation/email-address-schema.ts:48]: We can say the same for Zod default messages. The purpose of this pull request is to add reusable zod custom schema. We can refine them in another PR, but don`t forget that email address schema message will be different based on his context. `Email address` vs `Confirm email address`. The base email validation schema is the same but the error messages will differ.

Comment (sebastien-comeau) [reply frontend/app/.server/validation/email-address-schema.ts:48]: I'm planning doing it in another PR. It defeats the purpose of this PR.

**Review by gregory-j-baker — APPROVED** [review]

File: frontend/app/.server/validation/email-address-schema.ts (hunk 2 above)
Comment (gregory-j-baker) [reply frontend/app/.server/validation/email-address-schema.ts:48]: I think that i18n error messages falls within the responsibility of this PR, but am tired of arguing so I'll merge it.
